opensrp-chw-hf/src/main/java/org/smartregister/chw/hf/actionhelper/MotherStatusActionHelper.java

```diff
@@ -0,0 +1,280 @@
+        String deliveryTime = JsonFormUtils.getFieldValue(jsonPayload, "delivery_time");
+        String labourOnsetDate;
+        String labourOnsetTime;
+        if (LDDao.getLabourOnsetDate(baseEntityId) != null && LDDao.getLabourOnsetTime(baseEntityId) != null) {
+            labourOnsetDate = LDDao.getLabourOnsetDate(baseEntityId);
+            labourOnsetTime = LDDao.getLabourOnsetTime(baseEntityId);
+        } else {
+            labourOnsetDate = LDDao.getAdmissionDate(baseEntityId);
+            labourOnsetTime = LDDao.getAdmissionTime(baseEntityId);
+        }
+
+        // Add a check here if delivery date is not yet filled
+        if (StringUtils.isNotBlank(deliveryDate) && StringUtils.isNotBlank(deliveryTime)) {
+
+            String labourOnsetDateTime = labourOnsetDate + " " + labourOnsetTime;
+            String deliveryDateTime = deliveryDate + " " + deliveryTime;
+
+            SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm", Locale.getDefault());
+
+            Date date1 = simpleDateFormat.parse(labourOnsetDateTime);
+            Date date2 = simpleDateFormat.parse(deliveryDateTime);
+
+            long msDiff = date2.getTime() - date1.getTime();
+
+            for (int i = 0; i < fields.length(); i++) {
+                if (fields.getJSONObject(i).getString(KEY).equalsIgnoreCase("labour_duration")) {
+                    fields.getJSONObject(i).put(VALUE, msDiff);
+                }
+            }
+        }
+    }
+
+    private void startActionForChild() {
+        for (int i = 0; i < numberOfChildrenBorn; i++) {
+            // Get visit details for each individual child
+            if (isEdit) {
+                Visit lastVisit = LDLibrary.getInstance().visitRepository().getLatestVisit(baseEntityId, "LND " + ordinal(i + 1) + " Newborn");
+
+                if (lastVisit != null) {
+                    details = org.smartregister.chw.ld.util.VisitUtils.getVisitGroups(LDLibrary.getInstance().visitDetailsRepository().getVisits(lastVisit.getVisitId()));
+                }
+            }
+
+            String title;
+            if (numberOfChildrenBorn == 1) {
+                title = MessageFormat.format(context.getString(R.string.ld_new_born_status_action_title), "");
+            } else {
+                title = MessageFormat.format(context.getString(R.string.ld_new_born_status_action_title), "of " + ordinal(i + 1) + " baby");
+            }
+            NewBornActionHelper actionHelper = new NewBornActionHelper(baseEntityId, delivery_date, delivery_time, numberOfChildrenBorn, status, ordinal(i + 1));
+            BaseLDVisitAction action = null;
+            try {
+                action = new BaseLDVisitAction.Builder(context, title)
+                        .withOptional(false)
+                        .withHelper(actionHelper)
+                        .withDetails(details)
+                        .withBaseEntityID(baseEntityId)
```

Comment:
@gosso22 I see this was also refactored. Was there a logical reason for this? This was a required implementation for tying the immediate newborn care event to the child